### PR TITLE
Fixed parts of issue #56

### PR DESCRIPTION
**Pull request by swayam0322**
Fixed parts of issue #56 under Docathon-24:

- [x]  Add doc for TOR102 torch.load without weights_only parameter is unsafe
- [x]  Add doc for TOR104 Use of non-public function
- [x]  Add doc for TOR201 Find and fix deprecated pretrained parameters in TorchVision models
- [x]  Add doc for TOR202 The transform v2.ToTensor() is deprecated
- [x]  Add doc for TOR203 Consider replacing 'import torchvision.models as models'
- [x] Add doc for TOR401 Detected DataLoader running with synchronized implementation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,7 +108,56 @@ Migration guide:
 This function is deprecated. Use the `torch.nn.attention.sdpa_kernel` context manager instead.
 
 Migration guide:
+
 Each boolean input parameter (defaulting to true unless specified) of `sdp_kernel` corresponds to a `SDPBackened`. If the input parameter is true, the corresponding backend should be added to the input list of `sdpa_kernel`.
 
+### TOR102 Unsafe use of function 
+
+#### torch.load
```
Comment (kit1980) [inline README.md:116]: There should not be a level 3 and level 4 header, as TOR102 is just one item.
See TOR003 for example.

File: README.md [hunk 2]
```diff
@@ -108,7 +108,56 @@ Migration guide:
...
+
+#### torch.utils.data._utils.collate.default_collate
+
+Public functions are well-documented and supported by the library maintainers and the use of the non-public function `torch.utils.data._utils.collate.default_collate` is discouraged as it can can change without notice in future versions, leading to potential breakage in your code.   
+
+Migration Guide:
+
+For better maintainability and compatibility, please use the public function `torch.utils.data.dataloader.default_collate` instead.
+
+### TOR201 Use of deprecated parameter
```
Comment (kit1980) [inline README.md:142]: There should not be a level 3 and level 4 header, as TOR102 is just one item.
See TOR003 for example.

The title should be specific for the issue, not generic.

File: README.md [hunk 3]
```diff
@@ -108,7 +108,56 @@ Migration guide:
...
+
+#### Model(pretrained=True)
+
+The parameter `pretrained` has been deprecated in TorchVision models since PyTorch version 1.12.0. The `weights` parameter should be used instead. 
+
+```
+model = SomeModel(weights='<path_or_identifier_to_weights>')
+```
+
+### TOR202 Use of deprecated function
```
Comment (kit1980) [inline README.md:152]: There should not be a level 3 and level 4 header, as TOR102 is just one item.
See TOR003 for example.

The title should be specific for the issue, not generic.

**Review by kit1980 — CHANGES_REQUESTED** [review]
Standalone code blocks (with 3 backtics) look out of place, please make them inline or maybe remove.

**Comment by swayam0322** [comment]
@kit1980 Made the necessary changes

File: README.md [hunk 4]
```diff
@@ -108,7 +108,45 @@ Migration guide:
...
+
 Each boolean input parameter (defaulting to true unless specified) of `sdp_kernel` corresponds to a `SDPBackened`. If the input parameter is true, the corresponding backend should be added to the input list of `sdpa_kernel`.
 
+### TOR102 Unsafe use of `torch.load` without weights only parameter
+
+The use of `torch.load` without the `weights_only` parameter is unsafe. Loading an untrusted pickle file may lead to the execution of arbitrary malicious code and potential security issues.
+
+Migration Guide:
+
+Explicitly set `weights_only=False` only if you trust the data you load and full pickle functionality is needed,
```
Comment (kit1980) [inline README.md:120]: Only part of the sentence?

**Review by kit1980 — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -108,7 +108,45 @@ Migration guide:
...
+
+### TOR203 Consider replacing `import torchvision.models as models` with `from torchvision import models`.
+
+Consider replacing `import torchvision.models as models` with `from torchvision import models` to improve clarity, maintainability, and adhere to best practices and reducing potential confusion with other modules or variables named `models`. 
+This can lead to namespace conflicts and explicit import style helps avoid such issues.
+
+### TOR401 Detected DataLoader running with synchronized implementation
+
+Running synchronized implementations on `DataLoader` can lead to loss in data loading performance, especially when dealing with large datasets. A viable solution is to set the `num_workers` parameter to be greater than 0 when initializing the DataLoader class. This would parallelize the loading operations and would significantly increase performance. 
 ## License
```
Comment (kit1980) [inline README.md:151]: Should be an empty line before a header.

File: README.md (hunk 4 above)
Comment (swayam0322) [reply README.md]: Added the remaining text

File: README.md [hunk 6]
```diff
@@ -108,7 +108,50 @@ Migration guide:
 This function is deprecated. Use the `torch.nn.attention.sdpa_kernel` context manager instead.
 
 Migration guide:
-Each boolean input parameter (defaulting to true unless specified) of `sdp_kernel` corresponds to a `SDPBackened`. If the input parameter is true, the corresponding backend should be added to the input list of `sdpa_kernel`.
+
+Each boolean input parameter (defaulting to true unless specified) of `sdp_kernel` corresponds to a `SDPBackened`. 
+If the input parameter is true, the corresponding backend should be added to the input list of `sdpa_kernel`.
```
Comment (kit1980) [inline README.md:113]: This sentence doesn't make sense to me. "added to the input list of `sdpa_kernel`" - but we are removing `sdp_kernel`.

File: README.md [hunk 7]
```diff
@@ -108,7 +108,50 @@ Migration guide:
...
+`torch.utils.data._utils.collate.default_collate` is discouraged as it can can change without notice in future versions, 
+leading to potential breakage in your code.   
+
+Migration Guide:
+
+For better maintainability and compatibility, please use the public function `torch.utils.data.dataloader.default_collate` instead.
+
+### TOR201 Parameter `pretrained` is deprecated, please use `weights` instead.
+
+The parameter `pretrained` has been deprecated in TorchVision models since PyTorch version 1.12.0. The `weights` parameter should be used instead.
```
Comment (kit1980) [inline README.md:139]: Doesn't make sense.

**Comment by kit1980** [comment]
Sorry, I'm going to close this PR as a lot of things doesn't make sense even on a cursory reading, and correcting this takes more effort than doing from scratch.

**Pull request closed without merging**